### PR TITLE
ReferenceGrid

### PR DESCRIPTION
**Pull request by herflis**

### CODE REVIEW

File: packages/sn-controls-react/src/fieldcontrols/ReferenceGrid/ReferenceGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+/**
+ * Interface for RefernceGrid properties
+ */
+export interface ReferenceGridProps<T extends GenericContent, K extends keyof T>
+  extends ReactClientFieldSettingProps<T, K>,
+    ReactClientFieldSetting<T, K>,
+    ReactReferenceGridFieldSetting<T, K> {}
+/**
+ * Interface for TagsInput state
```
Comment (B3zo0) [inline packages/sn-controls-react/src/fieldcontrols/ReferenceGrid/ReferenceGrid.tsx]:
```suggestion
 * Interface for RefernceGrid state
```

File: packages/sn-controls-react/src/fieldcontrols/ReferenceGrid/ReferenceGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+> {
+  /**
+   * constructor
+   * @param {object} props
+   */
+  constructor(props: ReferenceGrid<T, K>['props']) {
+    super(props)
+    /**
+     * @type {object}
+     * @property {string} value default value
```
Comment (B3zo0) [inline packages/sn-controls-react/src/fieldcontrols/ReferenceGrid/ReferenceGrid.tsx]: I'm not sure about this comment. IMO it is not needed.

File: packages/sn-controls-react/src/fieldcontrols/ReferenceGrid/ReferenceGridFieldSettings.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/**
+ * @module FieldControls
+ *
+ */ /** */
```
Comment (B3zo0) [inline packages/sn-controls-react/src/fieldcontrols/ReferenceGrid/ReferenceGridFieldSettings.ts]:
```suggestion
 */
```

File: packages/sn-controls-react/src/fieldcontrols/ReferenceGrid/ReferenceGridFieldSettings.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { GenericContent } from '@sensenet/default-content-types'
+import { ReactReferenceFieldSetting } from '../ReferenceFieldSetting'
+
+/**
+ * Interface for TextareaFieldSetting properties
```
Comment (B3zo0) [inline packages/sn-controls-react/src/fieldcontrols/ReferenceGrid/ReferenceGridFieldSettings.ts]:
```suggestion
 * Interface for ReactReferenceGridFieldSetting properties
```

File: examples/sn-react-component-docs/stories/FieldControl.stories.tsx [hunk 5]
```diff
@@ -803,6 +811,78 @@ storiesOf('FieldControls.RadioButtonGroup', module)
...
+        onChange={action('change')}
+        name="Name"
+        dataSource={referenceGridDataSource}
+        repository={testRepository}
+        data-repository={testRepository}
+        data-selectionRoot={['/workspaces']}
+        data-allowedTypes={['File']}
+      />
+    ),
+    { notes: { markdown: nameNotes } },
```
Comment (gallayl) [inline examples/sn-react-component-docs/stories/FieldControl.stories.tsx]:
```suggestion
    { notes: { markdown: referenceGridNotes } },
```

File: examples/sn-react-component-docs/stories/FieldControl.stories.tsx [hunk 6]
```diff
@@ -803,6 +811,78 @@ storiesOf('FieldControls.RadioButtonGroup', module)
...
+            Avatar: { Url: 'https://dmsservice.demo.sensenet.com/Root/Sites/Default_Site/demoavatars/alba.jpg' },
+          } as User,
+        ]}
+        onChange={action('change')}
+        name="Name"
+        dataSource={tagsInputDataSource}
+        repository={testRepository}
+      />
+    ),
+    { notes: { markdown: nameNotes } },
```
Comment (gallayl) [inline examples/sn-react-component-docs/stories/FieldControl.stories.tsx]:
```suggestion
    { notes: { markdown: referenceGridNotes } },
```

**Pull request merged**